src/Core/Migration/V6_7/Migration1728040169AddPrimaryOrderDeliveryAndTransaction.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class Migration1728040169AddPrimaryOrderDeliveryAndTransaction extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1728040169;
+    }
+
+    public function update(Connection $connection): void
+    {
+        // No foreign key set both from order -> (primary) order delivery and order -> (primary) order transaction on
+        // purpose so the DAL can handle the circular reference. We have a similar situation with the order and order
+        // address.
+        if (!$this->columnExists($connection, 'order', 'primary_order_delivery_id')) {
+            $connection->executeStatement(
+                'ALTER TABLE `order`
+                ADD COLUMN `primary_order_delivery_id` BINARY(16) NULL DEFAULT NULL AFTER `language_id`,
+                ADD UNIQUE INDEX `uidx.order.primary_order_delivery` (`id`, `version_id`, `primary_order_delivery_id`);'
+            );
+            $connection->executeStatement(
+                'UPDATE `order`
```

Comment:
@mitelg @shyim What do you think about customers of millions of orders? Is this the right solution?

Reply:
updates on tables should be done in loops to prevent failures on big tables. also no limit and offset should be used. There must be some examples in other migrations for that

Reply:
Wait, for updating a million rows, you want a million single SQL updates? Or what to you mean with "in a loop" but "without limit / offset"?

Reply:
sorry, you can have a limit, but not an offset. so you loop the update with a reasonable limit until the `executeStatement` method does not return affected rows anymore. 

see https://github.com/shopware/shopware/blob/trunk/src/Core/Migration/V6_5/Migration1691057865UpdateSalutationDefaultForCustomer.php